pkg/build/generator/generator.go

```diff
@@ -218,9 +218,11 @@ func (g *BuildGenerator) Instantiate(ctx kapi.Context, request *buildapi.BuildRe
 	if err != nil {
 		return nil, err
 	}
-
+	if request.Binary == nil && bc.Spec.Source.Dockerfile == nil && bc.Spec.Source.Git == nil && len(bc.Spec.Source.Images) == 0 {
```

Comment:
(and the associated confusion of "what error do we return to a user when buildrequest validation failed because we got an spurious/temporary error retrieving the buildconfig associated with the buildrequest")